src/components/Sidebar.vue

```diff
@@ -36,43 +31,74 @@
   </div>
 </template>
 
-<script setup lang="ts">
+<script lang="ts">
+import { defineComponent, computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { i18n } from '../modules/i18n'
+
 import ViewListIcon from '/@vite-icons/heroicons-outline/view-list.vue'
 import CogIcon from '/@vite-icons/heroicons-outline/cog.vue'
 import HomeIcon from '/@vite-icons/heroicons-outline/home.vue'
 
-import { useI18n } from 'vue-i18n'
-import { isDark } from '/~/logics'
+export default defineComponent({
+  setup() {
+    const { t, availableLocales, locale } = useI18n()
 
-const { t, availableLocales, locale } = useI18n()
+    const toggleLocales = () => {
+      const locales = availableLocales
+      locale.value = locales[(locales.indexOf(locale.value) + 1) % locales.length]
```

Comment:
We may probably want to use a more straightforward method to change the locale. A ternary could be enough, but taking the next item in an array seems to me a bit “magic”.